### ISSUE TITLE
Outlook window steals focus from message windows when Task Pane closes

### ISSUE THREAD

**Issue by johnhrvatin-material** [OfficeDev/office-js#2880]
This is a 100% repro and new in the last 2-4 months.

## Your Environment
* Platform [PC desktop, Mac, iOS, Office on the web]: PC desktop 
* Host [Excel, Word, PowerPoint, etc.]: Outlook
* Office version number: MS365 Apps for enterprise - Outlook Version 2209 (Build 15629.201156 Click-to-Run)
* Operating System: Windows 10 Pro, 19044.2006
* Browser (if using Office on the web): n/a

## Expected behavior
The message window retains focus unless the user clicks away

## Current behavior
The main Outlook window steals focus without user interaction, moving the message window to the background

## Steps to reproduce
1. User clicks an add-ins toolbar button that opens a task pane
2. The task pane opens an email in a new window, and that window has focus
3. The task pane closes without the user doing anything

## Link to live example(s)
I don't have a test environment that's generally accessible but can work with someone on that if needed.

Video: https://www.loom.com/share/7bf41c9a6d9b4cef9cc326957f0e884a 

# Provide additional details
This is new in the last few months

## Context
This breaks our product's core user experience for our customers using Outlook desktop on Windows. Our product protects sensitive messages and this bug affects how they retrieve those messages.

**Comment by Oleg-O** [comment]
I was able to reproduce and without add-ins involved. Thanks for reporting this issue. It has been put on our backlog. We unfortunately have no timelines to share at this point.

Internal tracking id: Office: 6818737

**Comment by johnhrvatin-material** [comment]
Hi Oleg - can you share how you were able to reproduce "without add-ins involved"? I'm glad you were able to reproduce it, but I just want to make sure we're talking about the same issue.

Thank you so much for jumping on this today!

**Comment by johnhrvatin-material** [comment]
In case it's helpful, I narrowed down which update introduced the bug.

I went back through [old Office updates](https://learn.microsoft.com/en-us/officeupdates/update-history-microsoft365-apps-by-date) and determined the bug was introduced in **Version 2201**.

- Does NOT repro on Version 2112 (Build 14729.20260)
- DOES repro on Version 2201 (Build 14827.20158)

**Comment by Oleg-O** [comment]
Hi, John, indeed, it may not be the same issue. I've observed that this happens on its own shortly after Outlook restart. All I do is restart Outlook, select a message, double click on it to bring the message in a new window and then observe. After a few seconds, that new message will go into the background on its own.

Unfortunately, I am unable to reproduce this with add-in flow you described with the following code, on the same Outlook Version 2209 (Build 15629.20156 Click-to-Run).

```
Office.context.mailbox.displayMessageForm(Office.context.mailbox.item.itemId);

function test() {
Office.context.ui.closeContainer();
}

setTimeout(test, 5000);
```

How exactly in your step 3 the add-in closes without user taking an action? I would like to try to match your steps as closely as possible.

**Comment by johnhrvatin-material** [comment]
Hi Oleg,

We're doing it the same way you are so we set up a live repro to help figure out what's different.

- Add the attached add-in to Outlook desktop on Windows (I had to zip it for GitHub to allow it)
- Select a message
- Click the 'Test' button in the toolbar

The task pane will open, it will open the selected message in a new window, and after 5s it will call `closeContainer()` and the Outlook window will jump to the foreground.

I'm sure you can find this via the add-in's XML, but to make it easy I pasted the URL loaded in the task pane below. You can view source in the browser and see that the code we're running is the same as your sample. Here's a [video of my repro](https://www.loom.com/share/70853c04dc414d67ae3dde6f29b57009).

Task pane URL: https://staging.on.stellarite.io/oa/test 

Let me know what you discover. I'm fascinated by what's going on here. And of course let me know if you need more from me. JuanBL also has my email if that's easier.

[material-test.xml.zip](https://github.com/OfficeDev/office-js/files/9730629/material-test.xml.zip)

**Comment by Oleg-O** [comment]
Great, this helps. I'm able to consistently repro this now. We'll look into it.

Again, internal tracking id: Office: 6818737

**Comment by Oleg-O** [comment]
We have fixed the bug and the fix should be available in build 16.0.15812.10000 or later. When an updated build is available depends on the release channel a customer is on.
